src/main/java/seedu/address/storage/JsonAdaptedSchedule.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.storage;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.commons.exceptions.InvalidEnumArgumentException;
+import seedu.address.model.schedule.Event;
+import seedu.address.model.schedule.Schedule;
+
+public class JsonAdaptedSchedule {
+    private final List<JsonAdaptedEvent> events = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonAdaptedSchedule} with the given schedule details.
+     */
+    @JsonCreator
+    public JsonAdaptedSchedule(@JsonProperty("schedule") List<JsonAdaptedEvent> events) {
+        if (events != null) {
+            this.events.addAll(events);
+        }
+    }
+
+    /**
+     * Converts a given {@code Schedule} into this class for Jackson use.
```

Comment:
I think you should mention that this is a constructor here too

Reply:
Actually this part was copied exactly from JsonAdaptedPerson from the original AB3, I don't think its necessary to change it. Unless we change it for all the other JsonAdapted classes as well?

Reply:
Maybe it's too late at night. Just looking at the code, I can't tell that it was a constructor straightaway. So I think we should change it for future proofing

Reply:
Now that I think about it again, this constructor is actually not meant to be initialised by the developer, it is meant for Jackson to serialise objects. In my opinion it would be kind of misleading to change it to constructor.

Reply:
I think we should still mention that it is a constructor that is used for Jackson